### PR TITLE
Filter out non-usable moves in the teambuilder

### PR DESCRIPTION
**Pull request by KrisXV**

### CODE REVIEW

File: js/search.js [hunk 1]
```diff
@@ -968,6 +968,7 @@
 						} else */ if (learnsetEntry.indexOf(gen) < 0) {
 							continue;
 						}
+						if (this.gen === 8 && BattleMovedex[moveid].isNonstandard === "Past") continue;
```
Comment (Zarel) [inline js/search.js]: `if (learnsetEntry.indexOf(gen) < 0) {` should cover this...

Comment (KrisXV) [reply js/search.js]: It doesnt

Comment (Zarel) [reply js/search.js]: Oh, of course it doesn't.

Comment (Zarel) [reply js/search.js]: My mistake.

Comment (Zarel) [reply js/search.js]: Well, we'll need to add an exception for NatDex formats.

Can we do that now, actually? Will make it easier later.

Comment (Zarel) [inline js/search.js]: Well, we'll need to add an exception for NatDex formats.

Can we do that now, actually? Will make it easier later.

Comment (KrisXV) [reply js/search.js]: I mean, it is possible, but I believe @TheImmortal said somewhere that illegal moves wouldn't be usable in NatDex metas, just the Pokemon. Correct me if im wrong.

Comment (Zarel) [reply js/search.js]: I was planning on including illegal moves, though...

Comment (Zarel) [reply js/search.js]: I was going to just blanket-allow `Past` in the NatDex meta.

Comment (KrisXV) [reply js/search.js]: do you know what the formats will be named? maybe if its like [Gen 8 NatDex] i can add a format based exception

Comment (Zarel) [reply js/search.js]: `[Gen 8] National Dex` is the tentative name

Comment (KrisXV) [reply js/search.js]: did

**Pull request merged**